src/mongo/db/pipeline/accumulator_percentile.cpp

```diff
@@ -0,0 +1,195 @@
+        double digest_max = input[maxName].getDouble();
+        double digest_min = input[minName].getDouble();
+        double digest_size = input[digestSizeName].getDouble();
+
+        std::vector<mongo::TDigest::Centroid> centroids;
+        for (const auto& centroid: digest_centroids.getArray()) {
+            centroids.push_back(mongo::TDigest::Centroid(centroid[meanName].getDouble(), centroid[weightName].getDouble()));
+        };
+
+        digest = digest.merge({mongo::TDigest(centroids, digest_sum, digest_count, digest_max, digest_min, digest_size), digest});
+        this->percentile = input[percentileName].getDouble();
+        return;
+    }
+
+    // Determining 'digest_size'
+    if (this->digest_size == 0){
+        if (input.getDocument()["digest_size"].missing())
+            {
+            this->digest_size = 1000;
+            }
+        else
+            {
+            this->digest_size = input.getDocument()["digest_size"].getDouble();
+            }
+    }
+
+    uassert(51300, "The 'percentile' should be present in the input document.",
+    !input.getDocument()["percentile"].missing());
+
+    uassert(51301, "The 'value' should be present in the input document.",
+    !input.getDocument()["value"].missing());
+
+    this->percentile = input.getDocument()["percentile"].getDouble();
+
+    Value input_value = input.getDocument()["value"];
+
+    switch (input_value.getType()) {
+        case NumberDecimal:
+        case NumberLong:
+        case NumberInt:
+        case NumberDouble:
+            values.push_back(input_value.getDouble());
+            break;
+        default:
+            dassert(!input_value.numeric());
+            return;
+    }
+    
+    if (any_input == false)
+    {
+        digest = mongo::TDigest(digest_size);
+        any_input = true;
+    }
+
+    if (values.size() == chunk_size){
+        _add_to_tdigest();
+        }
+}
+
+intrusive_ptr<Accumulator> AccumulatorPercentile::create(
+    const boost::intrusive_ptr<ExpressionContext>& expCtx) {
+    return new AccumulatorPercentile(expCtx);
+}
+
+Value AccumulatorPercentile::getValue(bool toBeMerged) {
+
+    // To add the remainders
+    if (not values.empty()){
+        _add_to_tdigest();
+        }
+
+    if (toBeMerged) {
+        std::vector<Document> centroids;
+
+        for (const auto& centroid:digest.getCentroids()) {
+            centroids.push_back(Document{
+                    {"mean", centroid.mean()},
+                    {"weight", centroid.weight()}
+                });
+        };
+        
+        return Value(
+            Document{
+                {"centroids", Value(centroids)},
+                {"sum", digest.sum()},
+                {"count", digest.count()},
+                {"max", digest.max()},
+                {"min", digest.min()},
+                {"percentile", percentile},
+                {"digest_size", digest_size}
+            }
+        );
+    }
+
+    if (digest.count() == 0)
```

Comment:
Could you check if `if (digest.empty())` does the same thing ?
Using the `empty` method looks more readable to me.